airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -0,0 +1,324 @@
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {"Accept": "application/json"}
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        try:
+            return super()._send_request(request, request_kwargs)
+        except requests.exceptions.HTTPError as e:
+            error_message = e.response.text
+            if error_message:
+                self.logger.error(f"Stream {self.name}: {e.response.status_code} {e.response.reason} - {error_message}")
+            raise e
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data = response.json()
+        for data_field in self.data_fields:
+            if data_field is not None:
+                data = data.get(data_field, [])
+        if isinstance(data, list):
+            data = data
+        elif isinstance(data, dict):
+            data = [data]
+
+        for record in data:
+            updated_at = record.get("updated_at", 0)
+            if updated_at:
+                record["updated_at"] = datetime.fromtimestamp(record["updated_at"]).isoformat()  # convert timestamp to datetime string
+            yield record
+
+        # wait for 3,6 seconds according to API limit
+        time.sleep(3600 / self.rate_limit)
+
+
+class IncrementalIntercomStream(IntercomStream, ABC):
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+        # This method is called once for each record returned from the API to compare the cursor field value in that record with the current state
+        # we then return an updated state object. If this is the first time we run a sync or no state was passed, current_stream_state will be None.
+        current_stream_state = current_stream_state or {}
+        current_stream_state_date = current_stream_state.get("updated_at", str(self.start_date))
+        latest_record_date = latest_record.get(self.cursor_field, str(self.start_date))
+        return {"updated_at": max(current_stream_state_date, latest_record_date)}
+
+
+class StreamMixin:
+    stream = None
+
+    def stream_slices(self, sync_mode, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        for item in self.stream(authenticator=self.authenticator).read_records(sync_mode=sync_mode):
+            yield {"id": item["id"]}
+
+        yield from []
+
+
+class Admins(IntercomStream):
+    """Return list of all admins.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-admins
+    Endpoint: https://api.intercom.io/admins
+    """
+
+    primary_key = "id"
```

Comment:
Make `primary_key` to be a part of `IntercomStream`, it passes up here anyway, but you will have less code inside your stream classes.